ChatSecure/Classes/Controllers/FileTransferManager.swift

```diff
@@ -140,6 +153,21 @@ public class FileTransferManager: NSObject, OTRServerCapabilitiesDelegate {
     
     // MARK: - Public Methods
     
+    // Resume downloads, i.e. look for media items that are partially downloaded and retry getting them. TODO - use ranges
+    @objc public func resumeDownloads() {
+        connection.asyncRead { (transaction) in
+            transaction.enumerateKeysAndObjects(inCollection: OTRMediaItem.collection, using: { (key, object, stop) -> Void in
```

Comment:
Enumerating every object in the collection will be extremely slow, and this will block the global write queue, which may have cascading consequences where we do blocking reads on a write connection. I think that a SecondaryIndex for OTRMediaItems should be used on `.transferProgress` before this can be merged.